kratos/geometries/nurbs_shape_function_utilities/nurbs_interval.h

```diff
@@ -0,0 +1,175 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Thomas Oberbichler
+//
+//  Ported from the ANurbs library (https://github.com/oberbichler/ANurbs)
+//
+
+#if !defined(KRATOS_NURBS_INTERVAL_H_INCLUDED )
+#define  KRATOS_NURBS_INTERVAL_H_INCLUDED
+
+// System includes
+#include <algorithm>
+#include <limits>
+#include <sstream>
+
+// External includes
+
+// Project includes
+
+namespace Kratos
+{
+
+/// Class for optimized use of intervals
+/*
+* Provides universal geometrical utiltity functions for the computation of
+* curve and surface NURBS/ B-Spline shape functions.
+*/
+class Interval
```

Comment:
I would use a more specific name, this could be anything

Reply:
This class is actually a helper class. It can be anything. Just defines some limits and provides certain functions.